osu.Game/Screens/Multiplayer/MultiUserPanel.cs

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Screens.Multiplayer
+{
+    public class MultiUserPanel : ClickableContainer, IStateful<MultiUserPanel.UserState>
+    {
+        private UserState state;
+        public UserState State
+        {
+            get { return state; }
+            set
+            {
+                if (state == value)
+                    return;
+
+                state = value;
+                switch (state)
+                {
+                    case UserState.Guest:
+                        statusString = "Hosting a Room";
+                        UpdatePanel(this);
+                        break;
+
+                    case UserState.Host:
+                        statusString = "Waiting in a Room";
+                        UpdatePanel(this);
+                        break;
+                }
+            }
+        }
+
+        private string userName;
+        private string statusString;
+        private OsuSpriteText userSprite;
+        private OsuSpriteText statusSprite;
+
+        public const int PANEL_HEIGHT = 100;
+        public const int STATUS_HEIGHT = 33;
+        public const float PANEL_WIDTH = 0.25f;
+
+        public void UpdatePanel(MultiUserPanel panel)
+        {
+            statusSprite.Text = statusString;
+        }
+
+        public MultiUserPanel(string user = "Unknown")
+        {
+            userName = user;
+
+            RelativeSizeAxes = Axes.X;
+            Height = PANEL_HEIGHT;
+            Width = PANEL_WIDTH;
+            Masking = true;
+            CornerRadius = 10;
+            BorderThickness = 0;
+            EdgeEffect = new EdgeEffect
+            {
+                Type = EdgeEffectType.Shadow,
+                Colour = Color4.Black.Opacity(40),
+                Radius = 5,
+            };
+            Children = new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = new Color4(34,34,34, 255),
+                },
+                new Box
+                {
+                    RelativeSizeAxes = Axes.X,
+                    Height = STATUS_HEIGHT,
+                    Anchor = Anchor.BottomCentre,
+                    Origin = Anchor.BottomCentre,
+                    Colour = new Color4(17, 136, 170, 128),
+                },
+                new Container //Avatar
```

Comment:
Avatars are implemented, you should probably store a ```User``` on this class, and instantiate a new ```Avatar(some_user)```